status/barcode.py

```diff
@@ -23,45 +23,89 @@ def get(self):
-            file_for_printing = self.request.files['file_to_print']
-        except KeyError: # form entry for file was empty
-            print("no file was given")
-        else: #form entry for file was not empty
-            #retrieve file and transform the contained text to string format with utf-8 encoding
-            linesToPrint = str(file_for_printing[0]['body'], encoding='utf-8')
-            #check if file is already a label format file
-            barcodeFileMatch = re.compile('^\^XA')
-            isBarcodeFile = re.search(barcodeFileMatch, linesToPrint)
-            if isBarcodeFile: # if the file is already in the correct format it can be immediately moved to print
-                for _ in range(copies): # loops over copies to print
-                    print_barcode(linesToPrint)
-            else: # file submitted is a text file
-                array_linesToPrint = linesToPrint.splitlines() # split into the different lines of the text file
-                for line in array_linesToPrint:
-                    createdLabel = makeBarcode(line, match_barcode(line, False))
-                    createdLabel_joined = '\n'.join(createdLabel)
+            # printing from file input
+            try: # figure out if a file was entered in the form
+                file_for_printing = self.request.files['file_to_print']
+            except KeyError: # form entry for file was empty
+                print("no file was given")
+            else: #form entry for file was not empty
+                #retrieve file and transform the contained text to string format with utf-8 encoding
+                linesToPrint = str(file_for_printing[0]['body'], encoding='utf-8')
+                #check if file is already a label format file
+                barcodeFileMatch = re.compile('^\^XA')
+                isBarcodeFile = re.search(barcodeFileMatch, linesToPrint)
+                if isBarcodeFile: # if the file is already in the correct format it can be immediately moved to print
                     for _ in range(copies): # loops over copies to print
-                        print_barcode(createdLabel_joined)
+                        print_barcode(linesToPrint)
+                else: # file submitted is a text file
+                    array_linesToPrint = linesToPrint.splitlines() # split into the different lines of the text file
+                    for line in array_linesToPrint:
+                        createdLabel = makeBarcode(line, match_barcode(line, False))
+                        createdLabel_joined = '\n'.join(createdLabel)
+                        for _ in range(copies): # loops over copies to print
+                            print_barcode(createdLabel_joined)
+
+            # printing from string input
+            text_for_printing = self.get_argument('text_to_print') # retrieves string from form
+            withBarcode = self.get_argument('print_with_barcode') #retrieves information is the label should be printed with barcodes
```

Comment:
If the hidden element is removed, you can change this as 

```suggestion
            if self.get_argument('print_with_barcode', '')
```